### PR TITLE
Improve validation for AzureSQLUser and AzureSQLManagerUser

### LINKED ISSUE #1331: Bug: Azure SQL user and managed user allow blank dbName
**Describe the bug**
Rather than preventing the user from being created, it tries to grant any roles in the master database.

**To Reproduce**
* Deploy an Azure SQL database
* Create a user or managed user with dbName = "" with the db_datawriter role

**Expected behavior**
Validation should prevent the user from being created.

### LINKED ISSUE #1337: Bug: MinLength validations on types aren't showing up on CRDs
**Describe the bug**
There are 59 `kubebuilder:validation:MinLength` validation annotations on types under `api/`, but the generated CRDs in `config/crd/bases/` only have 4 occurences of `minLength` constraints in the schema. As an example, this code in `api/v1alpha1/cosmosdb_types.go`:
```go
type CosmosDBSpec struct {
	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
	// Important: Run "make" to regenerate code after modifying this file

	// +kubebuilder:validation:MinLength=0

	Location string `json:"location,omitempty"`
	// +kubebuilder:validation:Pattern=^[-\w\._\(\)]+$
	// +kubebuilder:validation:MinLength:1
	// +kubebuilder:validation:Required
	ResourceGroup          string                        `json:"resourceGroup"`
	Kind                   CosmosDBKind                  `json:"kind,omitempty"`
	Properties             CosmosDBProperties            `json:"properties,omitempty"`
	VirtualNetworkRules    *[]CosmosDBVirtualNetworkRule `json:"virtualNetworkRules,omitempty"`
	KeyVaultToStoreSecrets string                        `json:"keyVaultToStoreSecrets,omitempty"`
	Locations              *[]CosmosDBLocation           `json:"locations,omitempty"`
	IPRules                *[]string                     `json:"ipRules,omitempty"`
}
```
...results in these schema snippets:
```yaml
            location:
              minLength: 0
              type: string
...
            resourceGroup:
              pattern: ^[-\w\._\(\)]+$
              type: string
```

**Expected behavior**
The CRDs should have minLength set on all the fields that are marked with the annotation.

### PR DESCRIPTION
**Pull request by matthchr**
- Added validation to ensure that the DBName (and other required fields) of SQLUser/SQLManagedUser were specified. Fixes #1331.
- Added a validating webhook to ensure that the DBName `master` is not used, as that name is reserved.
- Added infrastructure in the tests so that webhooks can be run.
- Fixed issue with some kubebuilder annotations using `:` rather than `=` causing them to be ignored. Fixes #1337.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: controllers/suite_test.go [hunk 1]
```diff
@@ -152,6 +154,10 @@ func setup() error {
 	// +kubebuilder:scaffold:scheme
 	k8sManager, err = ctrl.NewManager(cfg, ctrl.Options{
 		Scheme: scheme.Scheme,
+		CertDir:            testEnv.WebhookInstallOptions.LocalServingCertDir,
+		Port:               testEnv.WebhookInstallOptions.LocalServingPort,
+		// TODO: Do we need to do the below - we have it in k8s-infra but don't seem to need it here.
```
Comment (Porges) [inline controllers/suite_test.go:159]: this is because we run multiple instances of envtest. if you're just running one then you won't get port conflicts

Comment (matthchr) [reply controllers/suite_test.go:159]: Thanks for the clarification, have removed this

File: config/default/webhookcainjection_patch.yaml [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
-apiVersion: admissionregistration.k8s.io/v1beta1
-kind: MutatingWebhookConfiguration
-metadata:
-  name: mutating-webhook-configuration
-  annotations:
-    cert-manager.io/inject-ca-from: $(NAMESPACE)/$(CERTIFICATENAME)
+# the variables $(NAMESPACE) and $(CERTIFICATENAME) will be substituted by kustomize.
+
+# NOTE: This section on mutating webhooks is commented out for now because we don't have any
+# and Kustomize doesn't like apply something to nothing
```
Comment (Porges) [inline config/default/webhookcainjection_patch.yaml:5]: 😆

**Review by theunrepentantgeek — APPROVED** [review]
Looks good to me.

**Pull request merged**